From e29503809327669dc8b74d3e320a455b2e53011a Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 30 Nov 2017 17:58:36 +0900
Subject: [PATCH 301/909] drm: rcar-du: Add external clk initialization for
 suspend to ram

commit 94d41ab694169d9cd6cb500b39721b1e6a84ccc5 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The suspend to ram execution turns off external clock register setting.
If clock value is not initialized, resetting in clk driver will not be
done at resume. This patch fixes this problem.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 69e9cdf572b4..e9e81d5f1d5d 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -371,6 +371,7 @@ static int rcar_du_pm_suspend(struct device *dev)
 {
 	struct rcar_du_device *rcdu = dev_get_drvdata(dev);
 	struct drm_atomic_state *state;
+	int i;
 
 	drm_kms_helper_poll_disable(rcdu->ddev);
 	drm_fbdev_cma_set_suspend_unlocked(rcdu->fbdev, true);
@@ -382,6 +383,9 @@ static int rcar_du_pm_suspend(struct device *dev)
 		return PTR_ERR(state);
 	}
 
+	for (i = 0; i < rcdu->num_crtcs; ++i)
+		clk_set_rate(rcdu->crtcs[i].extclock, 0);
+
 	rcdu->suspend_state = state;
 
 	return 0;
-- 
2.17.1

